src/controllers/howDoYouWantToPay.controller.ts

```diff
@@ -0,0 +1,145 @@
+import { StatusCodes } from 'http-status-codes'
+import { inject } from 'inversify'
+import { controller, httpGet, httpPost, requestBody } from 'inversify-express-utils'
+import { RedirectResult } from 'inversify-express-utils/dts/results'
+import { v4 as uuidv4 } from 'uuid'
+
+import BaseController from 'app/controllers/base.controller'
+import ApplicationStatus from 'app/models/dto/applicationStatus.enum'
+import DissolutionGetResponse from 'app/models/dto/dissolutionGetResponse'
+import PaymentType from 'app/models/dto/paymentType.enum'
+import HowDoYouWantToPayModel from 'app/models/form/howDoYouWantToPay.model'
+import Optional from 'app/models/optional'
+import DissolutionSession from 'app/models/session/dissolutionSession.model'
+import ValidationErrors from 'app/models/view/validationErrors.model'
+import { HOW_DO_YOU_WANT_TO_PAY_URI, PAY_BY_ACCOUNT_URI, SEARCH_COMPANY_URI } from 'app/paths'
+import { howDoYouWantToPaySchema } from 'app/schemas/howDoYouWantToPay.schema'
+import DissolutionService from 'app/services/dissolution/dissolution.service'
+import PaymentService from 'app/services/payment/payment.service'
+import SessionService from 'app/services/session/session.service'
+import TYPES from 'app/types'
+import FormValidator from 'app/utils/formValidator.util'
+
+interface ViewModel {
+  data?: Optional<HowDoYouWantToPayModel>
+  errors?: Optional<ValidationErrors>
+}
+
+@controller(HOW_DO_YOU_WANT_TO_PAY_URI)
+export class HowDoYouWantToPayController extends BaseController {
+
+  public constructor(
+    @inject(SessionService) private session: SessionService,
+    @inject(FormValidator) private validator: FormValidator,
+    @inject(DissolutionService) private service: DissolutionService,
+    @inject(PaymentService) private paymentService: PaymentService,
+    @inject(TYPES.PAY_BY_ACCOUNT_FEATURE_ENABLED) private PAY_BY_ACCOUNT_FEATURE_ENABLED: number
+  ) {
+    super()
+  }
+
+  @httpGet('')
+  public async get(): Promise<string | RedirectResult> {
+    const token: string = this.session.getAccessToken(this.httpContext.request)
+    const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
+
+    if (!this.PAY_BY_ACCOUNT_FEATURE_ENABLED || await this.isAlreadyPaid(dissolutionSession, token)) {
+      return this.redirect(SEARCH_COMPANY_URI)
+    }
```

Comment:
@matthewdoreilly Please confirm what should happen when the feature is not enabled. I thought that we should show an error, similar to how a 404 behaves.

Reply:
Yep, if the feature toggle is OFF then we should treat this page as if it doesn't exist (and hasn't ever existed), so we should return a 404.
And then if the application is already paid, then we do the redirect logic.

To show 404 error, we could do something like this,
```
@httpGet('')
  public async get(_: Request, response: Response, _2: NextFunction): Promise<string|void> {
    if (!this.PAY_BY_ACCOUNT_FEATURE_ENABLED) {
      return response.status(StatusCodes.NOT_FOUND).end()
    }
    ... // rest of code
```